quickwit/quickwit-indexing/src/actors/indexer.rs

```diff
@@ -114,14 +119,36 @@ impl IndexerState {
         partition_id: u64,
         last_delete_opstamp: u64,
         splits: &'a mut FnvHashMap<u64, IndexedSplitBuilder>,
+        other_split_opt: &'a mut Option<IndexedSplitBuilder>,
+        counter: &'a mut IndexerCounters,
         ctx: &ActorContext<Indexer>,
     ) -> anyhow::Result<&'a mut IndexedSplitBuilder> {
+        let num_splits = splits.len();
         match splits.entry(partition_id) {
             Entry::Occupied(indexed_split) => Ok(indexed_split.into_mut()),
             Entry::Vacant(vacant_entry) => {
-                let indexed_split =
-                    self.create_indexed_split_builder(partition_id, last_delete_opstamp, ctx)?;
-                Ok(vacant_entry.insert(indexed_split))
+                if num_splits as u32 + 1u32 >= self.max_num_partitions.get() {
```

Comment:
I'm hesitating on the "+1". Let's say a user knows that he has exactly 100 partitions. We will have the warnings with 100 partitions exactly.